ddtrace/tracer.py

```diff
@@ -331,6 +333,23 @@ def configure(
         if (collect_metrics is None and runtime_metrics_was_running) or collect_metrics:
             self._start_runtime_worker()
 
+        info = debug.collect(self)
```

Comment:
@brettlangdon by logging here we will log when the global tracer configures itself which is done when ddtrace is imported.

The issue here is that:

1) It will be done most likely before any log handlers are installed
2) If the user depends upon calling `configure` to reconfigure the global tracer to point at a different agent, then an `ERROR` will be logged for the initial configure call.


I don't see a way to avoid this unless we consider doing something crazy like waiting a fixed time duration before logging.

We could also decrease the level to `WARN` and encourage users to use the environment variables for configuration if possible.

Reply:
This is resolved by:

1. Reducing the log-level to WARNING
2. Allowing the diagnostic-level log messages to be opted out of with the environment variable

Not ideal since it'll still be a problem for users, but at least they can opt-out of it. 😕